### DISCUSSION TITLE
setQueryData creates a loop

### DISCUSSION THREAD

**Discussion by NotNotRama** [TanStack/query#2828] [Q&A]
I'm trying to fetch some data and immediately update said cache data on success, but if I use `setQueryData` it creates a loop.
```javascript
function App() {
  const queryClient = useQueryClient();
  const { isLoading, error, data } = useQuery('repoData', fetcher, {
    onSuccess: () => {
       queryClient.setQueryData('repoData', (oldData) => oldData);
    },
  });

  return (
    <div className="App">
      <Home />
    </div>
  );
}
```
For testing purposes, I tried returning the same data and I still get the same loop. I also tried using `setQueryData` outside `useQuery` without much success.

**Comment by TkDodo** [comment]
I’ve answered the question on StackOverflow: https://stackoverflow.com/questions/69701355/setquerydata-not-updating-the-cache-with-react-query
